sktime/transformations/series/simple_fourier.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Simple Fourier transform for time series, interfacing numpy.fft"""
+
+__author__ = ["blazingbhavneek"]
+
+
+import numpy as np
+import pandas as pd
+from numpy.fft import fft, fftfreq
+from sktime.transformations.base import BaseTransformer
+
+
+class simple_fourier(BaseTransformer):
```

Comment:
Minor grievance: name it `SimpleFourier` to stick to the class naming convention (although the linter might automatically fix it).

Reply:
@blazingbhavneek, adding to that, out of curiosity why is this called "simple fourier" rather than just "fourier transform"?

Reply:
Done! @achieveordie 
@KishManani It's a simple Fourier transform because it just interfaces numpy.fft, it doesn't take anything else as an input (and also, it assumes data is sampled uniformly, I raised this issue, but @fkiraly told me to do the simple interfacing task for now and do the other stuff later)

Reply:
Thanks @blazingbhavneek! Are there other Fourier transform implementations that you think could be added, other than an implementation which can handle non-uniformly sampled data? If this implementation is going to handle the majority of use cases, would it be worth just naming it as `FourierTransform`? I'm just mildly concerned if the namespace will have a lot of Fourier Transform implementations whether it's easy for a user to find what they're looking for (e.g., we already have `FourierFeatures`, this would add `SimpleFourier`, in the future we might have `NonUniformFourier` etc.). @fkiraly do you have any opinions on this? I know `sktime` now accepts multiple implementations of the same algorithm, is there any guidance or thought about the tedious topic of naming these 😅?

Reply:
hm, my gut feeling says: I would only have one file, e.g., `fourier_transform`, for any variations on the topic of fourier transform, and keep adding classes - or options to existing classes - unless the scope grows too large.

Fourier *features* are related but conceptually different in how they are applied, so one could argue either way sensibly, for separate or same module.

Reply:
>is there any guidance or thought about the tedious topic of naming these

I'm not too good at naming. Names should be descriptive and make clear to the user, optimally without inspection of the docstring, what a function, variable, class, does. The risk is that names get too long.

E.g., `SimpleFourier` is less clear than `FourierTransform`, because what does "simple" mean, and a user might ask, well, Fourier *what*?

Reply:
regarding applicability, I think we should try to extend the scope of the fourier transform transformer to irregular data etc in the same class.